test-commons/src/main/java/com/sonarsource/checks/verifier/NoncompliantCommentParser.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SonarQube Analyzer Test Commons
+ * Copyright (C) 2017-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.checks.verifier;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+import static com.sonarsource.checks.verifier.PreciseLocationParser.LINE_ADJUSTMENT;
+
+public final class NoncompliantCommentParser {
+
+  private static final Pattern START_WITH_NONCOMPLIANT = Pattern.compile("^(?i) *+Noncompliant");
+
+  private static final Pattern NON_COMPLIANT_REGEX = Pattern.compile("(?i) *+Noncompliant" + LINE_ADJUSTMENT
+  // issue count, ex: 2
+    + "(?: ++(?<issueCount>\\d++))?"
+    + " *+"
+    // messages, ex: {{msg1}} {{msg2}}
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)